### PR TITLE
Various Fixes 

### LINKED ISSUE #34: Sidebar Item select twice
Sidebar, click one, right location, click again, wrong location

### LINKED ISSUE #39: SunPy logo or text should be on mobile navbar.

### LINKED ISSUE #41: Using Sphinx 1.7 renders the sidebar on the website homepage.

### LINKED ISSUE #43: Mouseover Highlight Issue
Sidebar before:
screenshot from 2018-02-16 11-55-50

Sidebar after mouseover:
screenshot from 2018-02-16 11-55-56

### LINKED ISSUE #45: Ndcube Dropdown

### LINKED ISSUE #46: Make the theme have default for documentation not sunpy.org
Currently in all the projects using the theme as documentation you have to have this:
```
html_sidebars = {'**': ['docsidebar.html']}
```
line in your `conf.py`.

I think it would make more sense to have this as the default (i.e. in the `conf.py` in this repo) and then have sunpy.org override that.

### LINKED ISSUE #47: Download Buttons Break

### PR DESCRIPTION
**Pull request by nabobalis**
Fixed #47 
Fixed #46 
Fixed #45 
Fixed #41 (IT DIDNT BUT IM GONNA CLOSE IT VIA THIS PR AND YOU CANT STOP ME)
Fixed #34 (Not 100%, still jumps but now doesn't cut off content)
Fixed #43 (Not sure how well this will work long term but alas)
Fixed #39

Other changes:

Restored scroll bar in all browsers.

### CODE REVIEW

File: sunpy_sphinx_theme/conf.py [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
+from jinja2 import Environment
+
+from sunpy_sphinx_theme import get_html_theme_path
 
 html_theme_path = get_html_theme_path()
 html_theme = "sunpy"
+html_static_path = [os.path.join(html_theme_path[0], html_theme, "static")]
 html_favicon = os.path.join(html_theme_path[0], html_theme, "static", "img", "favicon-32.ico")
 
+sunpy_website_url_base = socket.gethostname()
```
Comment (Cadair) [inline sunpy_sphinx_theme/conf.py]: interesting. I would want to check this with github pages to actually make sure it returns sunpy.org

File: sunpy_sphinx_theme/sunpy/docsidebar.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{% set toctree = toctree(maxdepth=4, collapse=True, includehidden=True) %}
+{% set toctree = toctree(maxdepth=1, collapse=True, includehidden=False) %}
```
Comment (Cadair) [inline sunpy_sphinx_theme/sunpy/docsidebar.html]: what's this do?!

File: sunpy_sphinx_theme/sunpy/navbar.html [hunk 3]
```diff
@@ -38,10 +38,10 @@
                   <a href="http://docs.sunpy.org/projects/irispy/en/latest/">IRISPy</a>
```
Comment (Cadair) [inline sunpy_sphinx_theme/sunpy/navbar.html]: I am wondering if we should pull IRISPy as well. At the very least it should be at the end.

File: sunpy_sphinx_theme/sunpy/docsidebar.html (hunk 2 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/sunpy/docsidebar.html]: There are some changes I plan to revert. This is was my playing with the sidebar.

File: sunpy_sphinx_theme/conf.py (hunk 1 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/conf.py]: Yeah, I need to see how it works on deployment.

File: sunpy_sphinx_theme/sunpy/theme.conf [hunk 4]
```diff
@@ -2,18 +2,22 @@
 inherit=bootstrap
 
 [options]
+# Settings from the bootstrap theme
 navbar_pagenav = False
 source_link_position = False
-bootswatch_theme = flatly
+bootswatch_theme = sandstone
```
Comment (Cadair) [inline sunpy_sphinx_theme/sunpy/theme.conf]: what's this do?

File: sunpy_sphinx_theme/sunpy/navbar.html (hunk 3 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/sunpy/navbar.html]: We can move it to the bottom

File: sunpy_sphinx_theme/sunpy/theme.conf (hunk 4 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/sunpy/theme.conf]: Me playing. I reversed this locally.

**Comment by Cadair** [comment]
I think this is worthy of a 1.1

File: tool.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from jinja2 import Environment, PackageLoader, meta
```
Comment (Cadair) [inline tool.py]: oo what's this do?!

Comment (nabobalis) [reply tool.py]: Its just what I found to look at variables in templates. Tho it doesn't like our templates.

**Comment by nabobalis** [comment]
Sure but I will need to rebase this bigly when its ready.

File: sunpy_sphinx_theme/__init__.py [hunk 6]
```diff
@@ -1,8 +1,6 @@
-""" SunPy Sphinx Theme """
+"""SunPy Sphinx Theme"""
 import os
 
-__version__ = "1.0.7"
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy_sphinx_theme/__init__.py]: It is in the setup.py now

File: sunpy_sphinx_theme/sunpy/navbar.html [hunk 7]
```diff
@@ -38,10 +38,10 @@
...
             		</li>
             		<li class="dropdown-submenu">
-                  <a href="http://docs.sunpy.org/projects/solarbextrapolation/en/latest/">SolarBExtrapolation</a>
+                  <a href="http://docs.sunpy.org/projects/ndcube">ndcube</a>
             		</li>
             		<li class="dropdown-submenu">
-                  <a href="http://docs.sunpy.org/projects/sunkit-sst/en/latest/index.html">sunkit-sst</a>
+                  <a href="http://docs.sunpy.org/projects/drms">drms</a>
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy_sphinx_theme/sunpy/navbar.html]: Is it not?! I will check.

File: sunpy_sphinx_theme/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Cadair) [reply sunpy_sphinx_theme/__init__.py]: it should also be accessible as `sunpy_sphinx_theme.__version__` once imported though.

File: setup.py [hunk 8]
```diff
@@ -1,8 +1,10 @@
 from setuptools import setup
 
+from sunpy_sphinx_theme import __version__
```
Comment (Cadair) [inline setup.py]: this will break `pip install sunpy-sphinx-theme` in a clean env (test by `python setupy egg_info`)

Comment (nabobalis) [reply setup.py]: FINE MY GOD ITS NEVER GOOD ENOUGH FOR YOU

File: sunpy_sphinx_theme/sunpy/navbar.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy_sphinx_theme/sunpy/navbar.html]: It should be there.

**Comment by nabobalis** [comment]
THE LINTER NEVER WORKED

**Comment by nabobalis** [comment]
I think as it stands, I am happy with it. 

Not quite everything but it will do for now.

**Comment by Cadair** [comment]
screenshot_20180416_150531

**Comment by Cadair** [comment]
looks like it has something todo with the `::before` tag in the youtube widget

**Comment by nabobalis** [comment]
Fixed it.

**Comment by nabobalis** [comment]
WHY GILES, WHY CIRCLECI NO UPLOAD

**Pull request merged**